spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -474,35 +499,39 @@ protected void handleResult(Object resultArg, Object request, @Nullable Acknowle
 				invocationResult.messageReturnType() :
 				this.messageReturnType;
 
-		if (result instanceof CompletableFuture<?> completable) {
+		if (monoPresent && result instanceof Mono<?> mono) {
+			if (acknowledgment == null || !acknowledgment.isOutOfOrderCommit()) {
+				this.logger.warn("Container 'Acknowledgment' must be async ack for Mono<?> return type " +
+						"(or Kotlin suspend function); otherwise the container will ack the message immediately");
+			}
+			result = mono.toFuture();
```

Comment:
This is not OK to re-assign method argument.
I don't remember SonarCube rule, but we definitely must not override them.
So, you have to introduce internal `CompletableFuture<?>` variable and deal with it instead.